internal/pkg/oasparser/envoyCodegen/routesWithClusters.go

```diff
@@ -173,47 +166,104 @@ func createCluster(address core.Address, clusterName string) v2.Cluster {
 				},
 			},
 		},
-		//Hosts:                []*core.Address{h},
 	}
-
+	//fmt.Println(h.GetAddress())
 	return cluster
 }
 
-func createRoute(HostUrl string, resourcePath string, clusterName string) v2route.Route {
-	//var targetRegex = "/"
-	route := v2route.Route{
-		Match: &v2route.RouteMatch{
-			PathSpecifier: &v2route.RouteMatch_SafeRegex{
-				SafeRegex: &envoy_type_matcher.RegexMatcher{
-					EngineType: &envoy_type_matcher.RegexMatcher_GoogleRe2{
-						GoogleRe2: &envoy_type_matcher.RegexMatcher_GoogleRE2{
-							MaxProgramSize: nil,
+func createRoute(HostUrl string, basepath string, resourcePath string, clusterName string) v2route.Route {
+	var fullPath = basepath + resourcePath
+	var route v2route.Route
+
+	routepath, isHavingPathparameter := GenerateRegex(fullPath)
+
+	RouteAction := &v2route.Route_Route{
+		Route: &v2route.RouteAction{
+			HostRewriteSpecifier: &v2route.RouteAction_HostRewrite{
+				HostRewrite: HostUrl,
+			},
+			ClusterSpecifier: &v2route.RouteAction_Cluster{
+				Cluster: clusterName,
+			},
+		},
+	}
+
+	if isHavingPathparameter {
+		route = v2route.Route{
+			Match: &v2route.RouteMatch{
+				PathSpecifier: &v2route.RouteMatch_SafeRegex{
+					SafeRegex: &envoy_type_matcher.RegexMatcher{
+						EngineType: &envoy_type_matcher.RegexMatcher_GoogleRe2{
+							GoogleRe2: &envoy_type_matcher.RegexMatcher_GoogleRE2{
+								MaxProgramSize: nil,
+							},
 						},
+						Regex: routepath,
 					},
-					Regex: resourcePath,
 				},
+				/*Headers: []*v2route.HeaderMatcher {
```

Comment:
What is the purpose of the commented code. If there's a reason shall we mention that in a comment? If not, let's remove this :)

Reply:
This is already removed